src/lib.rs

```diff
@@ -197,16 +193,16 @@ pub(crate) const DEAD_STATE_IDX: u32 = 1;
 #[derive(Clone, Copy, PartialEq, Eq)]
 struct State {
     base: u32,
-    fach: u32,
-    output_pos: u32,
+    fail: u32,
+    opos_ch: u32,
```

Comment:
```suggestion
    // 3 bytes for output_pos and 1 byte for check.
    opos_ch: u32,
```

Reply:
updated in https://github.com/daac-tools/daachorse/pull/32/commits/38bd75ba7ca92dff699ca6461f11f14ab8621171.